lib/documentation.dart

```diff
@@ -18,76 +18,41 @@ import 'services/dartservices.dart';
 import 'src/util.dart';
 
 class DocHandler {
-  static const List cursorKeys = [
+  static const Set<int> cursorKeys = {
     KeyCode.LEFT,
     KeyCode.RIGHT,
     KeyCode.UP,
     KeyCode.DOWN
-  ];
+  };
 
   final Editor _editor;
   final Context _context;
 
   final NodeValidator _htmlValidator = PermissiveNodeValidator();
 
+  int /*?*/ _previousDoc;
```

Comment:
nit: Since we have a lot of things named "doc" in this codebase, a name like `_previousHash` or `_previousDocResultHash` would help with readability.